File: Reverse words in a given string - GFG/reverse-words-in-a-given-string.java
//{ Driver Code Starts
import java.util.*;
import java.lang.*;
import java.io.*;
class GFG {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t > 0) {
            String s = sc.next();
            Solution obj = new Solution();
            System.out.println(obj.reverseWords(s));
            t--;
        }
    }
}

// } Driver Code Ends



class Solution 
{
    void reverse(StringBuilder sb, int left, int right){
        while(left<=right){
            char temp = sb.charAt(left);
            sb.setCharAt(left, sb.charAt(right));
            sb.setCharAt(right, temp);
            left++;
            right--;
        }
    }
    String reverseWords(String s)
    {
        StringBuilder sb = new StringBuilder(s);
        sb.reverse();
        int n = s.length();
        int start = 0;
        for(int i=0;i<n;i++){
            if(sb.charAt(i)=='.'){
                reverse(sb, start, i-1);
                start = i+1;
            }
        }
        reverse(sb, start, n-1);
        return sb.toString();
    }
}

File: Reverse words in a given string - GFG/reverse-words-in-a-given-string.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution
{
    public:
    //Function to reverse words in a given string.
    string reverseWords(string s) 
    { 
        int n = s.size();
        int start = 0;
        reverse(s.begin(), s.end());
        for(int i=0;i<n;i++){
            if(s[i]=='.'){
                reverse(s.begin()+start, s.begin()+i);
                start=i+1;
            }
        }
        reverse(s.begin()+start, s.end());
        return s;
    } 
};

//{ Driver Code Starts.
int main() 
{
    int t;
    cin >> t;
    while (t--) 
    {
        string s;
        cin >> s;
        Solution obj;
        cout<<obj.reverseWords(s)<<endl;
    }
}
// } Driver Code Ends